client/hotrod-client/src/main/java/org/infinispan/client/hotrod/impl/transport/netty/ChannelPool.java

```diff
@@ -74,6 +74,32 @@ public void acquire(ChannelOperation callback) {
          callback.cancel(address, new RejectedExecutionException("Pool was terminated"));
          return;
       }
+
+      // We could acquire an active channel and submit the callback.
+      if (executeDirectlyIfPossible(callback)) return;
+
+      // wait action
+      if (maxWait > 0) {
+         TimeoutCallback timeoutCallback = new TimeoutCallback(callback);
+         timeoutCallback.timeoutFuture = executor.schedule(timeoutCallback, maxWait, TimeUnit.MILLISECONDS);
+         callback = timeoutCallback;
+      }
+
+      // Between the check time and adding the callback to the queue, we could have a channel available.
+      // Let's just try again.
+      if (!executeOrEnqueue(callback)) {
+         boolean remove = false;
+         try {
+            remove = executeDirectlyIfPossible(callback);
```

Comment:
Commenting this line will cause the behavior to be effective as before, and the `CloseBeforeEnqueuingTest` will fail with a timeout since the operation was never executed.